### PR TITLE
Fixes #33597 - add an experimental banner on host details page

### PR DESCRIPTION
**Pull request by amirfefer**
This banner specifies that the page is under development and also use as a call for feedback.

Screen Shot 2021-09-30 at 0 29 35

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js
Comment (ekohl) [inline webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js]: I'm not a huge fan of `feel free` but struggle to find a better phrasing. Perhaps @melcorr has some idea.

File: webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js
Comment (ekohl) [inline webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js]: Should it be an action?
```suggestion
              {__('Share feedback')}
```

File: webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js
Comment (melcorr) [inline webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js]: Since I left docs, I find myself saying 'feel free' more than I **feel** is necessary 
```suggestion
            'What do you think of new design? Send us your thoughts so we can continue to improve this page.'
```

How about a more direct approach ^

**Comment by amirfefer** [comment]
Thanks, @ekohl , @melcorr ! 
updated

File: webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js
Comment (melcorr) [inline webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js]:
```suggestion
            'What do you think of the new design? Send us your thoughts so we can continue to improve this page.'
```
I'm an idiot. Sorry @amirfefer - I caught Czechlish.

Comment (amirfefer) [reply webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js]: fixed :)

Comment (melcorr) [reply webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js]: Thank you  !

**Review by melcorr — DISMISSED** [review]
It reads well, but I've no idea if it works. So this approval is a "I approve this message" kind of approval :o)

**Review by ekohl — DISMISSED** [review]
Judging from the screenshot and text it looks all right to me but I'm not up to date with the JS review. @amirfefer shall I merge this or do you want some JS dev to review it?

**Comment by amirfefer** [comment]
@ekohl Thanks, I'd like to get an ACK from @MariSvirik before merging

**Comment by MariSvirik** [comment]
Hey @amirfefer , here are two options: 

1. Dismissable [inline alert](https://www.patternfly.org/v4/components/alert/design-guidelines)
Screenshot 2021-09-30 at 15 22 59

2. [Banner](https://www.patternfly.org/v4/components/banner/design-guidelines)
Screenshot 2021-09-30 at 15 22 48

PF4 doesn't recommend using toast notification for this use case.

**Comment by MariSvirik** [comment]
Based on the discussion we had in the host review meeting, I'm adding a design with an expandable inline alert with 2 actions 'Share feedback' and 'Switch to the old page'
Screenshot 2021-09-30 at 16 12 22

Screenshot 2021-09-30 at 16 24 11

But I think we don't need that description and therefore an expandable alert.

**Comment by amirfefer** [comment]
@MariSvirik, Thanks!
It seems that the expendable alert is a new feature of PF4 (merged just a few days ago), so foreman doesn't have this feature yet.
 How does it look without it? 
Screen Shot 2021-09-30 at 18 08 20

**Comment by MariSvirik** [comment]
@amirfefer 

If the users dismiss this alert is there a way to give us feedback or switch to the old version? 
I ran phrasing by the content team and their response was this:

> Experimental feature
This page redesign is still open for further evolution. Share your feedback to help us improve it, or switch back to familiar territory with its previous version.
Share your feedback              Switch to previous version

So we can either use alert with description (but that would be annoyingly large) or omit the description until we can have an expandable alert and for that phrasing would be (according to the content team):

> This page redesign is experimental and open for further evolution
Share your feedback              Switch to previous version

**Comment by ekohl** [comment]
Just wondering: is it stored that you can close the banner or is it shown again on every page? If it's not stored, I think I'd leave out the close button.

**Comment by amirfefer** [comment]
@MariSvirik
No description:
Screen Shot 2021-10-03 at 16 26 50

with description:
Screen Shot 2021-10-03 at 16 29 16

The second one looks a bit  annoyingly large, so I'd pick the first one

> If the users dismiss this alert is there a way to give us feedback or switch to the old version?

When the user re-enters the page (i.e refresh) this alert is visible again (we can hide it for the entire session by keeping the state in the session storage), we can also add buttons for feedback / switching back in the kebab dropdown:
Screen Shot 2021-10-03 at 16 34 59

@ekohl,
>  is it stored that you can close the banner or is it shown again on every page?

It's not stored so on every page reload it will pop again, we can use the session storage so it will be hidden for the entire session.

**Comment by amirfefer** [comment]
updated
- [x] use the alert with no description
- [x] add `share feedback` and `switch to prev version` in kebab dropdown

File: webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js]: Can this be in a scss file?

File: webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js]: I think it would make sense to open it in a new tab so the user can see what they give feedback on

File: webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js:52]: Isn't a host page with no id an error?

File: webpack/assets/javascripts/react_app/components/HostDetails/ActionsBar/index.js
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/HostDetails/ActionsBar/index.js]: This just renders a `<button` with an `href` attribute, so the link isn't working

**Review by MariaAga — CHANGES_REQUESTED** [review]
Great idea to add this!
Can the banner be aligned with the kebab? 
Screenshot from 2021-10-06 12-18-01

**Comment by MariSvirik** [comment]
> Great idea to add this! Can the banner be aligned with the kebab? Screenshot from 2021-10-06 12-18-01

@MariaAga @amirfefer kebab button should be 24 px from the right edge of the page.

File: webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js
Comment (MariSvirik) [reply webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js]: > It should open a new tab. (I thought @amirfefer presented it that way on our sync)

File: webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js
Comment (amirfefer) [reply webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js:52]: the first rendering doesn't contain `hostId`

**Comment by amirfefer** [comment]
Thanks @MariaAga, @MariSvirik !
done :)

Screen Shot 2021-10-06 at 21 57 00

File: webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js
Comment (ekohl) [inline webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js]: How about this?
```suggestion
        'This page redesign is experimental and under active development.'
```

Comment (amirfefer) [reply webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js]: I don't have a strong opinion, both look good to me.

Comment (MariSvirik) [reply webpack/assets/javascripts/react_app/components/HostDetails/ExperimentalAlert.js]: I like that

**Review by MariaAga — APPROVED** [review]
Looks like the text is approved and all the buttons work

**Comment by MariaAga** [comment]
Thanks @amirfefer @ekohl @MariSvirik !

**Comment by ezr-ondrej** [comment]
CP 3.0: #8842

**Pull request merged**
